src/app/qgswmsdimensiondialog.cpp

```diff
@@ -0,0 +1,234 @@
+  QVariant refValue;
+  if ( mNameComboBox->findText( name ) != -1 )
+  {
+    refValue = mReferenceValueComboBox->currentData();
+  }
+  return QgsVectorLayerServerProperties::WmsDimensionInfo( name, mFieldComboBox->currentField(),
+         mEndFieldComboBox->currentField(),
+         mUnitsLineEdit->text(), mUnitSymbolLineEdit->text(),
+         mDefaultDisplayComboBox->currentData().toInt(), refValue );
+}
+
+void QgsWmsDimensionDialog::nameChanged( const QString &name )
+{
+  // reset input
+  mUnitsLabel->setEnabled( true );
+  mUnitsLineEdit->setEnabled( true );
+  mUnitsLineEdit->clear();
+  mUnitSymbolLabel->setEnabled( true );
+  mUnitSymbolLineEdit->setEnabled( true );
+  mUnitSymbolLineEdit->clear();
+
+  // Is the name a predefined value?
+  if ( mNameComboBox->findText( name ) != -1 )
+  {
+    QString data = mNameComboBox->currentData().toString();
+    if ( data.compare( QLatin1String( "time" ) ) == 0 ) // Time
+    {
+      mFieldComboBox->setFilters( QgsFieldProxyModel::String | QgsFieldProxyModel::DateTime );
+      mEndFieldComboBox->setFilters( QgsFieldProxyModel::String | QgsFieldProxyModel::DateTime );
+      mUnitsLineEdit->setText( QStringLiteral( "ISO8601" ) );
+      mUnitsLabel->setEnabled( false );
+      mUnitsLineEdit->setEnabled( false );
+      mUnitSymbolLabel->setEnabled( false );
+      mUnitSymbolLineEdit->setEnabled( false );
+    }
+    else if ( data.compare( QLatin1String( "elevation" ) ) == 0 ) // Elevation
+    {
+      mFieldComboBox->setFilters( QgsFieldProxyModel::Numeric );
+      mEndFieldComboBox->setFilters( QgsFieldProxyModel::Numeric );
+    }
+  }
+  else
+  {
+    mFieldComboBox->setFilters( QgsFieldProxyModel::AllTypes );
+    mEndFieldComboBox->setFilters( QgsFieldProxyModel::AllTypes );
+  }
+
+  /*QStringList predefinedUpperNames;
+  for ( const QString &n : QgsVectorLayerServerProperties::predefinedWmsDimensionNames() )
+  {
+    predefinedUpperNames << n.toUpper();
+  }
+  const QString upperName = name.toUpper();
+
+  if ( predefinedUpperNames.contains( upperName ) )
+  {
+    if ( upperName.compare( QLatin1String( "TIME" ) ) == 0 ) // Time
+    {
+      mFieldComboBox->setFilters( QgsFieldProxyModel::String | QgsFieldProxyModel::DateTime );
+      mEndFieldComboBox->setFilters( QgsFieldProxyModel::String | QgsFieldProxyModel::DateTime );
+      mUnitsLineEdit->setText( QStringLiteral( "ISO8601" ) );
+      mUnitsLabel->setEnabled( false );
+      mUnitsLineEdit->setEnabled( false );
+      mUnitSymbolLabel->setEnabled( false );
+      mUnitSymbolLineEdit->setEnabled( false );
+    }
+    else if ( upperName.compare( QLatin1String( "ELEVATION" ) ) == 0 ) // Elevation
+    {
+      mFieldComboBox->setFilters( QgsFieldProxyModel::Numeric );
+      mEndFieldComboBox->setFilters( QgsFieldProxyModel::Numeric );
+    }
+  }
+  else
+  {
+    mFieldComboBox->setFilters( QgsFieldProxyModel::AllTypes );
+    mEndFieldComboBox->setFilters( QgsFieldProxyModel::AllTypes );
+  }*/
```

Comment:
Can we remove this commented code block?

Reply:
thanks to point it